src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -319,14 +322,49 @@ private String generateAssertionsEntryPointClassContent(final Set<ClassDescripti
         ? determineBestEntryPointsAssertionsClassPackage(classDescriptionSet)
         : entryPointClassPackage;
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, PACKAGE, classPackage);
-
+    Set<String> imports;
+    if (entryPointAssertionsClassContent.contains(IMPORTS)) {
+      imports = extractImports(classDescriptionSet);
+      entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, listNeededImports(imports));
+    } else {
+      imports = new HashSet<>();
+    }
     String allEntryPointsAssertionContent = generateAssertionEntryPointMethodsFor(classDescriptionSet,
-                                                                                  entryPointAssertionMethodTemplate);
+                                                                                  entryPointAssertionMethodTemplate,
+                                                                                  imports);
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, ALL_ASSERTIONS_ENTRY_POINTS,
                                                allEntryPointsAssertionContent);
     return entryPointAssertionsClassContent;
   }
 
+  private static String listNeededImports(Collection<String> imports) {
+    StringBuilder builder = new StringBuilder();
+    for (String anImport : imports) {
+      builder.append(format(IMPORT_LINE, anImport, LINE_SEPARATOR));
+    }
+    return builder.toString();
+  }
+
+  private static Set<String> extractImports(Set<ClassDescription> classDescriptionSet) {
+    Set<String> imports = new TreeSet<>();
+    Map<String, String> importedQualifiedName = new HashMap<>(classDescriptionSet.size() * 2);
+    for (ClassDescription description : new TreeSet<>(classDescriptionSet)) {
+      String outerClassName = description.getOuterClassName();
+      if (!importedQualifiedName.containsKey(outerClassName)) {
+        String toImport = description.getPackageName() + "." + outerClassName;
+        imports.add(toImport);
+        importedQualifiedName.put(outerClassName, toImport);
+      }
+
+      if (!importedQualifiedName.containsKey(simpleAssertClassName(description))) {
+        String toImport = fullyQualifiedAssertClassName(description);
+        imports.add(toImport);
+        importedQualifiedName.put(simpleAssertClassName(description), toImport);
+      }
+    }
+    return imports;
```

Comment:
is that imports = importedQualifiedName.values() ? if so we can get rid of the imports set